### PR TITLE
refactor: action handling

### PR DESCRIPTION
**Pull request by micielski**
After this is done, there should be only one and ergonomic way to interact with Transmission Client without locking any mutexes

- [x] Remove Tick from actions
- [x] Get rid of mutexes
- [x] Make Task status work again

### CODE REVIEW

**Comment by aidanaden** [comment]
im just curious.. what was the reasoning behind having `Arc<Mutex>` in the first place? and why did u decide to revert from it afterwards?

**Comment by micielski** [comment]
> what was the reasoning behind having `Arc<Mutex>` in the first place?

It was one way of making this app multithreaded

> why did u decide to revert from it afterwards?

This is just another way to make the threads communicate with each other but with channels, and I think this one reads better (and theoretically should be faster, but not much).

**Comment by micielski** [comment]
TLDR biggest changes:
- Components now can only return either ComponentAction::Nothing or ComponentAction::Quit in order to talk with its parents instead of Action
- Added a new enum "UpdateAction" which is used in place of the gone mutexes now
- Action::Tick is gone, in exchange App::tick() is being called every 250ms (which calls all of its childs that implement tick())
- Transmission's action handler now returns a value via a oneshot channel to its caller instead of using mutexes (if there's anything to return of course)

I hope I didn't miss anything
